### PR TITLE
Added gapit status.

### PR DESCRIPTION
**Pull request by AWoloszyn**
This will allow gapit to attach to a running server and
provides a status update in your terminal window.

### CODE REVIEW

File: cmd/gapit/status.go [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+		children   map[uint64]*tsk
+	}
+
+	ancestors := make(map[uint64][]uint64)
+	activeTasks := make(map[uint64]*tsk)
+	totalBlocked := 0
+
+	var findTask func(*map[uint64]*tsk, []uint64) *tsk
+
+	findTask = func(base *map[uint64]*tsk, indices []uint64) *tsk {
```
Comment (Qining) [inline cmd/gapit/status.go]: Probably no need to make it a pointer to map?

Comment (AWoloszyn) [reply cmd/gapit/status.go]: Fair, I constantly get confused by golang semantics for maps, as they are non-obvious.

File: gapis/service/service.go [hunk 2]
```diff
@@ -134,6 +135,9 @@ type Service interface {
 	// This is a debug API, and may be removed in the future.
 	Profile(ctx context.Context, pprof, trace io.Writer, memorySnapshotInterval uint32) (stop func() error, err error)
 
+	// Status starts resolving status events. It calls f for every update and m for every status update.
```
Comment (Qining) [inline gapis/service/service.go]: Comment: m for every memory status update?

Comment (AWoloszyn) [reply gapis/service/service.go]: Fixed

File: core/app/status/task.go [hunk 3]
```diff
@@ -113,11 +126,35 @@ func Start(ctx context.Context, name string, args ...interface{}) context.Contex
...
+		background: false,
+	}
+	log.D(ctx, "Starting task: %s", t.name)
+
+	parent.add(t)
+	onTaskStart(ctx, t)
+	return PutTask(ctx, t)
+}
+
+// Start returns a new context for a long running task.
```
[the remark this answers is not in the document]
Comment (AWoloszyn) [reply core/app/status/task.go]: Fixed

File: cmd/gapit/status.go [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+			f(next_tsk)
+			if len(indices) == 1 {
+				return
+			}
+			forAncestors(&next_tsk.children, indices[1:], f)
+		}
+	}
+
+	var print func(*map[uint64]*tsk, int, bool)
+	print = func(m *map[uint64]*tsk, d int, background bool) {
```
[the remark this answers is not in the document]
Comment (AWoloszyn) [reply cmd/gapit/status.go]: Fixed

File: cmd/gapit/status.go [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+			}
+			return findTask(&next_tsk.children, indices[1:])
+		}
+		return nil
+	}
+
+	var forAncestors func(*map[uint64]*tsk, []uint64, func(*tsk))
+	forAncestors = func(base *map[uint64]*tsk, indices []uint64, f func(*tsk)) {
+		if next_tsk, ok := (*base)[indices[0]]; ok {
+			f(next_tsk)
```
[the remark this answers is not in the document]
Comment (AWoloszyn) [reply cmd/gapit/status.go]: Fixed

File: cmd/gapit/status.go (hunk 1 above)
Comment (AWoloszyn) [reply cmd/gapit/status.go]: Fixed

**Pull request merged**